File: Python/AD-HOC/3058.py
qtd = int(input())
valores = {}
menor = []

while qtd > 0:
  linha = input().split(' ')
  valor = float(linha[0])
  gramas = int(linha[1])

  valores[valor] = gramas
  qtd -= 1

for i in valores: menor.append(i * 1000 / valores[i])

menor.sort()
menor = menor[0]
print(f"{menor:.2f}")

File: C++/AD-HOC/3058.cpp
#include <stdio.h>
#include <iostream>
#include <vector>
#include <algorithm>

int main() {

  int qtd;
  double valor, gramas;
  std::vector<double> valores;
  std::cin >> qtd;
  
  while(qtd > 0){
    std::cin >> valor >> gramas;
    valor *= 1000 / gramas;
    valores.push_back(valor);
    qtd--;
  }

  std::sort(valores.begin(), valores.end());

  printf("%.2lf\n", valores[0]);

}
